design/design.md

```diff
@@ -0,0 +1,54 @@
+
+# CS124 Lab 1 Design Document
+#### Team Members: Roman Herrera and Kobe Lin
+
+
+## Design Decisions
+One of the first design decisions that we made was to make the page itself look similar to a notepad. This was to make using the webpage to be analagous to writing to-do notes down on paper. We chose colors typical to most notepads in order to mirror this decision. Another way we implemented this idea was by having the actual body of the to-do list be left-justified, reflecting the convention of English to be written and read from left-to-right. If we were to implement this for languages read and written from right-to-left, we would have instead made our items be right-justified. In fact, we decided to have the items on our page be left-aligned in general, as that is where users will likely scan their eyes first. Furthermore, we felt that to-do lists should be very clean and easy to look at, so we chose to have the general look of the webpage be very minimal so as not to feel too distracting for the user. 
+
+![lab1base1.png](https://www.dropbox.com/s/86qqvf1ja9krs9u/lab1base1.png?dl=0&raw=1)
+
+Another decision we made was about buttons. 
+- In regards to placement, we felt that it would make the most sense to have the "Create new item" button be right underneath the list itself, and have newly added items appear right above it. As for the "Rename", "Show Uncompleted", and "Delete All Completed" buttons, we decided to have it grouped together at the bottom of the page as they each can affect multiple list items at once, and thus have somewhat similar functionality. 
+- We also decided that when a button is "in use", the buttons would stay in the same place but have their text modified to show that an action is tak  ing place (for ex. the Rename button changes its text to "Done" upon click, and back to "Rename" once the user clicks it again).
+
+![lab1renaming1.png](https://www.dropbox.com/s/pc7jkufqhezqvkk/lab1renaming1.png?dl=0&raw=1)
+
+
+## Alternative Designs
+
+When designing our webpage, there were 4 alternative designs that we considered.
+
+##### 1. Putting the rename/show uncompleted/delete all completed buttons in the top right
+![alttopright.jpg](https://www.dropbox.com/s/4b1v6jszsq0rvei/alttopright.jpg?dl=0&raw=1)
+One of our initial considerations was potentially putting these buttons in the top right of the page. However, we decided against this because we felt that it would be better for the body of the to-do list to be the first thing under the header. We would rather the user's attention to immediately be locked onto their to-do list rather than the functionality buttons. 
+
+##### 2. Making the add button a "+" instead of the word "Add"
+![altplus.jpg](https://www.dropbox.com/s/2l2xx30okl0bgno/altplus.jpg?dl=0&raw=1)

```

Comment:
The three buttons change location on the screen as there are more or fewer todo items. What will happen if there are more items than will fit on the screen?  Would it be better if the buttons were in a fixed location so the user can easily find them?